src/management/moderation/components/moderation-offer.component.js

```diff
@@ -0,0 +1,155 @@
+'use strict';
+
+/**
+ * @ngdoc component
+ * @name udb.search.directive:udbSearchBar
+ * @description
+ * # udbQuerySearchBar
+ */
+angular
+  .module('udb.management.moderation')
+  .component('udbModerationOffer', {
+    templateUrl: 'templates/moderation-offer.html',
+    controller: ModerationOfferComponent,
+    controllerAs: 'moc',
+    bindings: {
+      offerId: '@',
+      offerType: '@'
+    }
+  });
+
+/* @ngInject */
+function ModerationOfferComponent(ModerationManager, jsonLDLangFilter, OfferWorkflowStatus, $uibModal) {
+  var moc = this;
+  var defaultLanguage = 'nl';
+
+  moc.loading = true;
+  moc.offer = {};
+  moc.sendingJob = false;
+  moc.error = false;
+
+  moc.isReadyForValidation = isReadyForValidation;
+  moc.isApproved = isApproved;
+  moc.isRejected = isRejected;
+  moc.approve = approve;
+  moc.askForRejectionReasons = askForRejectionReasons;
+
+  // fetch offer
+  ModerationManager
+    .getModerationOffer(moc.offerId)
+    .then(function(offer) {
+      offer.updateTranslationState();
+      moc.offer = jsonLDLangFilter(offer, defaultLanguage);
+    })
+    .catch(showLoadingError)
+    .finally(function() {
+      moc.loading = false;
+    });
+
+  function showLoadingError(problem) {
+    showProblem(problem || {title:'Dit aanbod kon niet geladen worden.'});
+  }
+
+  function isReadyForValidation() {
+    return moc.offer.workflowStatus === OfferWorkflowStatus.READY_FOR_VALIDATION;
+  }
+
+  function isApproved() {
+    return moc.offer.workflowStatus === OfferWorkflowStatus.APPROVED;
+  }
+
+  function isRejected() {
+    return moc.offer.workflowStatus === OfferWorkflowStatus.REJECTED;
+  }
+
+  function approve() {
+    moc.sendingJob = true;
+    moc.error = false;
+    ModerationManager
+      .approve(moc.offer)
+      .then(function() {
+        moc.offer.workflowStatus = OfferWorkflowStatus.APPROVED;
+      })
+      .catch(showProblem)
+      .finally(function() {
+        moc.sendingJob = false;
+      });
+  }
+
+  function askForRejectionReasons() {
+    var modalInstance = $uibModal.open({
+      templateUrl: 'templates/reject-offer-confirm-modal.html',
+      controller: 'RejectOfferConfirmModalCtrl'
+    });
+
+    modalInstance.result.then(reject);
```

Comment:
'k zou dat kunnen, de reden waarom k t niet gedaan heb is omdat ik alle calls naar de moderation service wou groeperen in deze, zodat we niet 1 call voor de approve hier hebben en 3 calls in de modal controller.